### PR TITLE
APPS-623 Add callee executables to caller applet's bundledDepends.

### PR DESCRIPTION
**Pull request by r-i-v-a**

### CODE REVIEW

File: compiler/src/test/scala/dx/compiler/CompilerTest.scala [hunk 1]
```diff
@@ -787,6 +787,69 @@ class CompilerTest extends AnyFlatSpec with Matchers with BeforeAndAfterAll {
...
+    val applet = dxApi.applet(appletId)
+    val desc = applet.describe(Set(Field.RunSpec))
+
+    desc.runSpec match {
+      case Some(rs) => {
+        val bundledDepends = rs.asJsObject.fields
+          .get("bundledDepends")
+          .getOrElse(throw new Exception(s"Expected ${appletId} to have bundledDepends"))
+        // Assert bundledDepends contains item with expected name
+        bundledDepends match {
```
Comment (r-i-v-a) [inline compiler/src/test/scala/dx/compiler/CompilerTest.scala:799]: Test still needs debugging.

File: compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala [hunk 2]
```diff
@@ -254,11 +257,24 @@ case class ApplicationCompiler(typeAliases: Map[String, Type],
...
 
+    // Add executables called by this applet to bundledDepends to ensure cloning
+    val bundledDependsExec: Vector[JsValue] = executableDict.map {
+      case (name, link) =>
+        JsObject(
+            Constants.BundledDependsNameKey -> JsString(name),
+            Constants.BundledDependsIdKey -> JsObject(
+                DxUtils.DxLinkKey -> JsString(link.dxExec.id)
+            ),
+            Constants.BundledDependsStagesKey -> JsArray(Vector.empty)
```
Comment (jdidion) [inline compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala:266]: Is this a placeholder for future changes?

**Review by jdidion — COMMENTED** [review]
LGTM - will approve once tests pass

File: compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala (hunk 2 above)
Comment (r-i-v-a) [reply compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala:266]: No, it's to specify that there are no stages / entrypoints of the executable for which the bundled object needs to be downloaded. Without that, the default is to always download it to the worker.

https://documentation.dnanexus.com/developer/api/running-analyses/io-and-run-specifications#run-specification @jdidion

**Review by jdidion — DISMISSED** [review]

**Pull request merged**